### PR TITLE
Fix bug when the query result is returned in first HTTP call

### PR DESCRIPTION
**Pull request by buremba**
If Trino returns the query result in the first HTTP call, the `columns` property is not set so the library throws the` Cursor must be loaded before call TrinoResultMetaData` exception.

### CODE REVIEW

**Comment by ebyhr** [comment]
@cla-bot check

**Comment by hashhar** [comment]
When does Trino return the result in the first request itself? I can't think of any case except where the query planning/analysis/scheduling fails. Just want to understand what was the user-facing behaviour.

**Review by hashhar — COMMENTED** [review]
The fix looks correct but I don't understand how the original error was being hit.

File: trino/client.py [hunk 1]
```diff
@@ -507,20 +507,23 @@ def execute(self, additional_http_headers=None) -> TrinoResult:
...
         self._stats.update({"queryId": self.query_id})
-        self._stats.update(status.stats)
+        self._update_stats(status)
         self._warnings = getattr(status, "warnings", [])
         if status.next_uri is None:
             self._finished = True
         self._result = TrinoResult(self, status.rows)
         return self._result
+    
+    def _update_stats(self, status):
```
Comment (findepi) [inline trino/client.py:517]: rename function, since it does more than just update stats

**Comment by hashhar** [comment]
@buremba Can you please squash all the commits (since it's a single logical change)?

**Review by hashhar — APPROVED** [review]
Thanks. Merged.

**Pull request merged**